src/merkle_tree.rs

```diff
@@ -1,6 +1,7 @@
 use sha3::{Digest, Sha3_256};
```

Comment:
I would run `cargo fmt` , in many places the code looks oddly formatted though it might be Github screwing things up.

Reply:
Noted. I've included "cargo-fmt on save" to avoid having this problem again